samples/matter/common/src/ota_multi_image_downloader.cpp

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "ota_multi_image_downloader.h"
+
+#include <dfu/dfu_target.h>
+#include <pm_config.h>
+
+size_t OTAMultiImageDownloaders::sCurrentId;
+
+std::array<OTAMultiImageDownloader, OTAMultiImageDownloaders::CurrentImageID::LIMIT>
+	OTAMultiImageDownloaders::sImageHandlers{
+		{ OTAMultiImageDownloader{ 0, PM_APP_1_CORE_APP_DOWNLOAD_OFFSET, PM_APP_1_CORE_APP_DOWNLOAD_SIZE,
+					   PM_APP_1_CORE_APP_ADDRESS, PM_APP_1_CORE_APP_SIZE },
+		  OTAMultiImageDownloader{ 1, PM_APP_1_CORE_NET_DOWNLOAD_OFFSET, PM_APP_1_CORE_NET_DOWNLOAD_SIZE,
+					   PM_APP_1_CORE_NET_ADDRESS, PM_APP_1_CORE_NET_SIZE },
+		  OTAMultiImageDownloader{ 2, PM_APP_2_CORE_APP_DOWNLOAD_OFFSET, PM_APP_2_CORE_APP_DOWNLOAD_SIZE,
+					   PM_APP_2_CORE_APP_ADDRESS, PM_APP_2_CORE_APP_SIZE },
+		  OTAMultiImageDownloader{ 3, PM_APP_2_CORE_NET_DOWNLOAD_OFFSET, PM_APP_2_CORE_NET_DOWNLOAD_SIZE,
+					   PM_APP_2_CORE_NET_ADDRESS, PM_APP_2_CORE_NET_SIZE } }
+	};
+
+static const device *sExtFlash = DEVICE_DT_GET(DT_CHOSEN(nordic_pm_ext_flash));
```

Comment:
kExtFlash?

Reply:
Btw does it need to be static?

Reply:
fyi: `const` implies `static` linkage in C/C++ so all consts are automatically static.

Reply:
Not sure about the C:
*"Names at the top-level namespace scope (file scope in C) that are const and not extern have external linkage in C, but internal linkage in C++. "* 
But yes, it yields the internal linkage in C++.

Reply:
I am however wondering about the naming convention that we tend to implicitly agree upon, for me k prefix is more suitable for numeric consts. But this is just a nit, maybe not worth discussing.

Reply:
oh, I didn't know C does it differently. Thanks!